qiskit_ibm_provider/utils/backend_converter.py

```diff
@@ -0,0 +1,155 @@
+    else:
+        for gate in configuration.gates:
+            name = gate.name
+            gate_props = (
+                {tuple(x): None for x in gate.coupling_map}  # type: ignore[misc]
+                if hasattr(gate, "coupling_map")
+                else {None: None}
+            )
+            gate_len = len(gate.coupling_map[0]) if hasattr(gate, "coupling_map") else 0
+            if name in name_mapping:
+                target.add_instruction(name_mapping[name], gate_props)
+            else:
+                custom_gate = Gate(name, gate_len, [])
+                target.add_instruction(custom_gate, gate_props)
+        target.add_instruction(Measure())
+    # parse global configuration properties
+    if hasattr(configuration, "dt"):
+        target.dt = configuration.dt
+    if hasattr(configuration, "timing_constraints"):
+        target.granularity = configuration.timing_constraints.get("granularity")
+        target.min_length = configuration.timing_constraints.get("min_length")
+        target.pulse_alignment = configuration.timing_constraints.get("pulse_alignment")
+        target.aquire_alignment = configuration.timing_constraints.get(
+            "acquire_alignment"
+        )
+    # If a pulse defaults exists use that as the source of truth
+    if defaults is not None:
+        inst_map = defaults.instruction_schedule_map
+        for inst in inst_map.instructions:
+            for qarg in inst_map.qubits_with_instruction(inst):
+                sched = inst_map.get(inst, qarg)
+                if inst in target:
+                    try:
+                        qarg = tuple(qarg)
+                    except TypeError:
+                        qarg = (qarg,)
+                    if inst == "measure":
+                        for qubit in qarg:
+                            target[inst][(qubit,)].calibration = sched
+                    else:
+                        target[inst][qarg].calibration = sched
+    return target
+
+
+def qubit_props_dict_from_props(
+    properties: BackendProperties,
+) -> Dict[int, IBMQubitProperties]:
+    """Uses BackendProperties to construct
+    and return IBMQubitProperties class.
+    """
+    qubit_props = {}
+    for qubit, _ in enumerate(properties.qubits):
+        t_1 = properties.t1(qubit) * 1e-6  # microseconds to seconds
+        t_2 = properties.t2(qubit) * 1e-6  # microseconds to seconds
+        frequency = properties.frequency(qubit) * 1e9  # GHz to Hz
+        try:
+            anharmonicity = (
+                properties.qubit_property(qubit, "anharmonicity")[0] * 1e9
+            )  # GHz to Hz
+        except Exception:  # pylint: disable=broad-except
+            anharmonicity = None
+        qubit_props[qubit] = IBMQubitProperties(  # type: ignore[no-untyped-call]
+            t1=t_1,
+            t2=t_2,
+            frequency=frequency,
+            anharmonicity=anharmonicity,
+        )
+    return qubit_props
```

Comment:
This should be changed to a list return instead of a dict with int keys.

Reply:
Oops sorry didn't notice these comments, sure will open a new PR with these changes.